examples/DataApi-Water-And-Energy-Usage/template/modules/data_l1/src/main/scala/com/my/currency/data_l1/CustomRoutes.scala

```diff
@@ -0,0 +1,94 @@
+package com.my.currency.data_l1
+
+import cats.data.OptionT
+import cats.effect.IO
+import cats.implicits.catsSyntaxOption
+import com.my.currency.shared_data.Data.deserializeState
+import com.my.currency.shared_data.Types.{UpdateUsageTransaction, UsageState}
+import com.my.currency.shared_data.Utils.customStateDeserialization
+import org.http4s._
+import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
+import org.http4s.dsl.io._
+import org.tessellation.currency.dataApplication.L1NodeContext
+import org.tessellation.schema.address.Address
+import io.circe.parser._
+
+object CustomRoutes {
+  private def getState(context: L1NodeContext[IO]) = {
+    OptionT(context.getLastCurrencySnapshot)
+      .flatMap(_.data.toOptionT)
+      .flatMapF(deserializeState(_).map(_.toOption))
+      .value
+  }
+
+  private def getMetagraphStateFromSnapshotOrdinal(ordinal: Long): Array[Byte] = {
+    val headers: Iterable[(String, String)] = Iterable(("Accept", "application/json"))
+    val response = requests.get(s"http://host.docker.internal:9400/snapshots/$ordinal", headers = headers)
```

Comment:
Is there an internal method we can use to fetch this instead of making an HTTP call?

Reply:
I don't know, @ryle-goehausen do you know if we have something like this?

Reply:
Just to clarify again from discussion only the L0 could in theory access this data, so on the L1 this is probably best way to accomplish this, route could potentially be moved